lib/logstash/inputs/http_poller.rb

```diff
@@ -49,19 +49,35 @@ class LogStash::Inputs::HTTP_Poller < LogStash::Inputs::Base
   def register
     @host = Socket.gethostname.force_encoding(Encoding::UTF_8)
 
-    @logger.info("Registering http_poller Input", :type => @type, :schedule => @schedule, :timeout => @timeout)
-
     setup_ecs_field!
     setup_requests!
   end
 
   # @overload
   def stop
+    shutdown_scheduler_and_close_client(:wait)
+  end
+
+  # @overload
+  def close
+    shutdown_scheduler_and_close_client
+  end
+
+  def shutdown_scheduler_and_close_client(opt = nil)
+    @logger.debug("closing http client", client: client)
+    begin
+      client.close # since Manticore 0.9.0 this shuts-down/closes all resources
+    rescue => e
+      details = { exception: e.class, message: e.message }
+      details[:backtrace] = e.backtrace if @logger.debug?
+      @logger.info "failed closing http client", details
```

Comment:
good call, since this is on `close` I think it would make sense to use warn level, as well:
```suggestion
      @logger.warn "failed closing http client", details
```